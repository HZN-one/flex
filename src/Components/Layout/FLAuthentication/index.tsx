import React, { memo } from "react";
import { Box, Grid } from "@mui/material";

import { FATypography } from "@Atoms";
import { FOHeader } from "@Organisms";

import { IFLAuthenticationProps } from "./FLAuthentication.interface";

export const FLAuthentication = memo((props: IFLAuthenticationProps) => {
  const { testID, children, header, isCenter, childrenGridWidth } = props;

  return (
    <React.Fragment>
      <FOHeader
        testID="header-authentication"
        headerType="auth"
        position="fixed"
        {...header}
      />
      <Grid
        data-testid={testID}
        component="main"
        container
        columnSpacing={3}
        rowSpacing={isCenter ? 0 : 2}
        justifyContent="center"
        alignItems={isCenter ? "center" : "flex-start"}
        sx={{
          flexGrow: 1,
          minHeight: theme =>
            `calc(100vh - ${theme.mixins.toolbar.minHeight}px - ${theme.spacing(
              9
            )})`,
          marginTop: theme => `${theme.mixins.toolbar.minHeight}px`,
        }}
      >
        <Grid item sm={childrenGridWidth}>
          {children}
        </Grid>
      </Grid>
      <Box component="footer" sx={{ py: 1, textAlign: "center" }}>
        <FATypography
          testID="typography-auth-layout-footer"
          color="textSecondary"
          variant="caption"
        >
          &copy; {new Date().getFullYear()} - PT. HZN Teknologi Indonesia
        </FATypography>
      </Box>
    </React.Fragment>
  );
});

FLAuthentication.displayName = "FLAuthentication";
