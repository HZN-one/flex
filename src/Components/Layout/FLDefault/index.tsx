import React, { memo } from "react";
import { Box } from "@mui/material";
import { FOHeader, FOSideBar } from "@Organisms";
import { IFLDefaultProps } from "./FLDefault.interface";

export const FLDefault = memo((props: IFLDefaultProps) => {
  const { testID, children, header, ...sidebarProps } = props;

  return (
    <Box sx={{ display: "flex" }}>
      <FOHeader
        testID="header-layout-default"
        headerType="default"
        position="fixed"
        {...header}
      />
      <FOSideBar testID="sidebar-layout-default" {...sidebarProps} />
      <Box
        data-testid={testID}
        component="main"
        sx={{
          flexGrow: 1,
          marginTop: theme => `${theme.mixins.toolbar.minHeight}px`,
        }}
      >
        {children}
      </Box>
    </Box>
  );
});

FLDefault.displayName = "FLDefault";
