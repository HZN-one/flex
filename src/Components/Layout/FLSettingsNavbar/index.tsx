import React, { memo } from "react";
import { Box, Container } from "@mui/material";

import type { ReactNode } from "react";
import { styled } from "@mui/material/styles";

// sidebar
import List from "@mui/material/List";
import ChevronLeftIcon from "@mui/icons-material/ChevronLeft";
import ListItem from "@mui/material/ListItem";
import ListItemIcon from "@mui/material/ListItemIcon";
import ListItemText from "@mui/material/ListItemText";
import SettingsIcon from "@mui/icons-material/Settings";

// appbar
import MenuIcon from "@mui/icons-material/Menu";

// PR: get from flex
import { FATypography } from "@Atoms/FATypography";
import { FAIconButton } from "@Atoms/FAIconButton";
import { FAButton } from "@Atoms/FAButton";

// main components
import { FOHeader } from "@Organisms/FOHeader";
import { FOSideBar } from "@Organisms/FOSideBar";

export interface IFLSettingsNavbarLayout {
  testID: `header-${string}`;
  title?: string;
  popUp?: ReactNode;
  children?: ReactNode;
  headbarPosition?: "fixed" | "absolute" | "relative" | "static" | "sticky";
  sections: {
    title: string;
    path: string;
    icon: JSX.Element;
    children?: {
      title: string;
      path: string;
    }[];
  }[];
}

const Main = styled("main", { shouldForwardProp: prop => prop !== "open" })<{
  open?: boolean;
}>(({ theme, open }) => ({
  transition: theme.transitions.create("margin", {
    easing: theme.transitions.easing.sharp,
    duration: theme.transitions.duration.leavingScreen,
  }),
  ...(open && {
    transition: theme.transitions.create("margin", {
      easing: theme.transitions.easing.easeOut,
      duration: theme.transitions.duration.enteringScreen,
    }),
    marginLeft: `${250}px`,
  }),
}));

export const FLSettingsNavbarLayout = memo((props: IFLSettingsNavbarLayout) => {
  const { testID, popUp, headbarPosition, sections, title } = props;
  const [open, setOpen] = React.useState(false);

  const initialPath = window.location.pathname;
  const pathMarkerSidebar = (path: string) => {
    if (initialPath === path) {
      return {
        bgcolor: "#FDEBEA",
        width: "100%",
        display: "flex",
        alignItems: "center",
        borderRadius: 16,
        padding: "0 12px",
      };
    }
    return {
      width: "100%",
      display: "flex",
      alignItems: "center",
      borderRadius: 16,
      padding: "0 12px",
    };
  };

  const handleDrawerOpen = () => {
    setOpen(true);
  };

  const handleDrawerClose = () => {
    setOpen(false);
  };

  return (
    <Box data-testid={testID}>
      <FOHeader
        position={headbarPosition ?? "static"}
        testID="header-settings"
        open={open}
        startAdornment={
          <>
            <FAIconButton
              testID="icon-button-layoutSettings"
              color="inherit"
              aria-label="open drawer"
              onClick={open ? handleDrawerClose : handleDrawerOpen}
              edge="start"
              sx={{ mr: 2 }}
            >
              <MenuIcon />
            </FAIconButton>
            <FATypography
              testID="typography-layoutSettings"
              variant="h6"
              noWrap
            >
              {title}
            </FATypography>
            <Box
              sx={{
                flexGrow: 1,
                ml: 2,
              }}
            />
            <Box sx={{ ml: 1 }}>{popUp}</Box>
          </>
        }
      ></FOHeader>
      <FOSideBar
        sx={{
          width: 250,
          flexShrink: 0,
          "& .MuiDrawer-paper": {
            width: 250,
            boxSizing: "border-box",
          },
        }}
        testID="side-bar-settings"
        open={open}
        sections={sections}
        footer={
          <List>
            <ListItem
              button
              onClick={() => window.location.assign("/settings")}
            >
              <Box sx={pathMarkerSidebar("/settings")}>
                <ListItemIcon>
                  <SettingsIcon />
                </ListItemIcon>
                <ListItemText primary={"Settings"} />
              </Box>
            </ListItem>
            <ListItem onClick={() => window.location.assign("/launchpad")}>
              <FAButton
                testID="button-layoutSettings"
                fullWidth
                sx={{
                  borderRadius: "8px",
                  color: "black",
                  borderColor: "black",
                  mt: 2,
                }}
                variant="outlined"
                startIcon={<ChevronLeftIcon />}
              >
                Back to Launchpad
              </FAButton>
            </ListItem>
          </List>
        }
      ></FOSideBar>
      <Main open={open}>
        <Container maxWidth="xl">{props.children}</Container>
      </Main>
    </Box>
  );
});

FLSettingsNavbarLayout.displayName = "FLSettingsNavbarLayout";
