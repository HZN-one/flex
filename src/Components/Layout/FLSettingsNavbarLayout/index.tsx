import React, { memo } from "react";
import { Box, Container } from "@mui/material";

import type { ReactNode } from "react";
import { styled } from "@mui/material/styles";

// sidebar
import List from "@mui/material/List";
import ListItem from "@mui/material/ListItem";
import ListItemIcon from "@mui/material/ListItemIcon";
import ListItemText from "@mui/material/ListItemText";
import SettingsIcon from "@mui/icons-material/Settings";

// appbar
import MenuIcon from "@mui/icons-material/Menu";

// get from flex
import { FATypography } from "@Atoms/FATypography";
import { FAIconButton } from "@Atoms/FAIconButton";
import { FAButton } from "@Atoms/FAButton";

// main components
import { FOHeader } from "@Organisms/FOHeader";
import { FOSideBar } from "@Organisms/FOSideBar";

export interface IFLSettingsNavbarLayout {
  testID: `header-${string}`;
  title?: string;
  popUp?: ReactNode;
  children?: ReactNode;
  headbarPosition?: "fixed" | "absolute" | "relative" | "static" | "sticky";
  sections: {
    title: string;
    path: string;
    icon: JSX.Element;
    children?: {
      title: string;
      path: string;
    }[];
  }[];
  handleLogOut?: () => void;
}

const Main = styled("main", { shouldForwardProp: prop => prop !== "open" })<{
  open?: boolean;
}>(({ theme, open }) => ({
  transition: theme.transitions.create("margin", {
    easing: theme.transitions.easing.sharp,
    duration: theme.transitions.duration.leavingScreen,
  }),
  ...(open && {
    transition: theme.transitions.create("margin", {
      easing: theme.transitions.easing.easeOut,
      duration: theme.transitions.duration.enteringScreen,
    }),
    marginLeft: `${250}px`,
  }),
}));

export const FLSettingsNavbarLayout = memo((props: IFLSettingsNavbarLayout) => {
  const { testID, popUp, handleLogOut, headbarPosition, sections, title } =
    props;
  const [open, setOpen] = React.useState(false);

  const initialPath = window.location.pathname;
  const pathMarkerSidebar = (path: string) => {
    if (initialPath === path) {
      return {
        bgcolor: "#FDEBEA",
        width: "100%",
        display: "flex",
        alignItems: "center",
        borderRadius: 16,
        padding: "0 12px",
      };
    }
    return {
      width: "100%",
      display: "flex",
      alignItems: "center",
      borderRadius: 16,
      padding: "0 12px",
    };
  };

  const handleDrawerOpen = () => {
    setOpen(true);
  };

  const handleDrawerClose = () => {
    setOpen(false);
  };

  return (
    <Box data-testid={testID}>
      <FOHeader
        position={headbarPosition ?? "static"}
        testID="header-settings"
        open={open}
        startAdornment={
          <>
            <FAIconButton
              testID="icon-button-layoutSettings"
              color="inherit"
              aria-label="open drawer"
              onClick={open ? handleDrawerClose : handleDrawerOpen}
              edge="start"
              sx={{ mr: 2 }}
            >
              <MenuIcon />
            </FAIconButton>
            <FATypography
              testID="typography-layoutSettings"
              variant="h6"
              noWrap
            >
              {title}
            </FATypography>
            <Box
              sx={{
                flexGrow: 1,
                ml: 2,
              }}
            />
            <Box sx={{ ml: 1 }}>{popUp}</Box>
          </>
        }
      ></FOHeader>
      <FOSideBar
        sx={{
          width: 250,
          flexShrink: 0,
          "& .MuiDrawer-paper": {
            width: 250,
            boxSizing: "border-box",
          },
        }}
        testID="side-bar-settings"
        open={open}
        sections={sections}
        footer={
          <List>
            <ListItem
              button
              onClick={() => window.location.assign("/settings")}
            >
              <Box sx={pathMarkerSidebar("/settings")}>
                <ListItemIcon>
                  <SettingsIcon />
                </ListItemIcon>
                <ListItemText primary={"Settings"} />
              </Box>
            </ListItem>
            <ListItem onClick={handleLogOut}>
              <FAButton
                testID="button-layoutSettings"
                fullWidth
                sx={{
                  borderRadius: "8px",
                  color: "black",
                  borderColor: "black",
                  mt: 2,
                }}
                variant="outlined"
                startIcon={
                  <svg
                    width="17"
                    height="16"
                    viewBox="0 0 17 16"
                    fill="none"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M11.833 4.66667L10.893 5.60667L12.613 7.33333H5.83301V8.66667H12.613L10.893 10.3867L11.833 11.3333L15.1663 8L11.833 4.66667ZM3.16634 3.33333H8.49967V2H3.16634C2.43301 2 1.83301 2.6 1.83301 3.33333V12.6667C1.83301 13.4 2.43301 14 3.16634 14H8.49967V12.6667H3.16634V3.33333Z"
                      fill="#221F20"
                    />
                  </svg>
                }
              >
                Log Out
              </FAButton>
            </ListItem>
          </List>
        }
      ></FOSideBar>
      <Main open={open}>
        <Container maxWidth="xl">{props.children}</Container>
      </Main>
    </Box>
  );
});

FLSettingsNavbarLayout.displayName = "FLSettingsNavbarLayout";
