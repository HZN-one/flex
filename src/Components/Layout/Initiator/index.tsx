import React, { FC } from "react";

import { ThemeProvider } from "@mui/material/styles";
import CssBaseline from "@mui/material/CssBaseline";
import { Props } from "./Initiator.interface";
// import { createTheme } from '@mui/material/styles';

import { baseTheme } from "../../..";

export const ThemeProviderWrapper: FC<Props> = function (props) {
  // const primaryPallete = replaceTheme?.palette?.primary;
  // const secondaryPallete = replaceTheme?.palette?.secondary;
  // const infoPallete = replaceTheme?.palette?.info;
  // const errorPallete = replaceTheme?.palette?.error;
  // const warningPallete = replaceTheme?.palette?.warning;
  // const successPallete = replaceTheme?.palette?.success;
  // const contrastThreshold = replaceTheme?.palette?.contrastThreshold;
  // const tonalOffset = replaceTheme?.palette?.tonalOffset;

  return (
    <ThemeProvider theme={baseTheme(props)}>
      <CssBaseline />
      {props.children}
    </ThemeProvider>
  );
};
