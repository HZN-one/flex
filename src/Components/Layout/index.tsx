export * from "./FLAuthentication";
export * from "./FLDefault";
export * from "./Initiator";
export * from "./FLSettingsNavbarLayout";
