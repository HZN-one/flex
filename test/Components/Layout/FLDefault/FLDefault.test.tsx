/**
 * @jest-environment jsdom
 */

import * as React from "react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";

import { FLDefault } from "@Components/Layout";

describe("FLDefault", () => {
  it("should match snapshot", () => {
    const { container } = render(
      <FLDefault
        testID="layout-default-test"
        header={{
          title: "Title",
          userAvatarSrc: "#",
          accountName: "Company Name",
        }}
        linkComponent="a"
        menu={[
          {
            title: "Dashboard",
            path: "/",
            icon: "dashboard",
          },
          {
            title: "Users",
            path: "/users",
            icon: "people",
            subMenu: [
              {
                title: "List",
                path: "/users/list",
              },
              {
                title: "Create",
                path: "/users/create",
              },
            ],
          },
        ]}
        footerMenu={[
          {
            title: "Settings",
            path: "/settings",
            icon: "settings",
          },
        ]}
        buttonLogout={{
          children: "Logout",
          onClick: () => {},
        }}
      >
        Children component
      </FLDefault>
    );
    expect(container).toMatchSnapshot();
  });
});
