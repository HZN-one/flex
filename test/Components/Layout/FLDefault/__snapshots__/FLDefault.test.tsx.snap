// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FLDefault should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-k008qs"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorInherit MuiAppBar-positionFixed mui-fixed css-koqw40-MuiPaper-root-MuiAppBar-root"
      data-testid="header-layout-default"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container css-1ugee5g-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto css-1wrgmsj-MuiGrid-root"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
              data-testid="typography-header-title"
            >
              Title
            </h5>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto css-1wrgmsj-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-1n2mv2k"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circular css-gfc8i8-MuiAvatar-root"
              >
                <img
                  class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                  src="#"
                />
              </div>
              <p
                class="MuiTypography-root MuiTypography-body1 css-yyvcc2-MuiTypography-root"
                data-testid="typography-header-company-name"
              >
                Company Name
              </p>
            </div>
          </div>
        </div>
      </div>
    </header>
    <div
      class="MuiDrawer-root MuiDrawer-docked css-7ik032-MuiDrawer-docked"
      data-testid="sidebar-layout-default"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-12i7wg6-MuiPaper-root-MuiDrawer-paper"
      >
        <nav
          class="MuiList-root MuiList-padding MuiList-subheader css-11tbbah-MuiList-root"
        >
          <div
            class="MuiListSubheader-root MuiListSubheader-sticky css-10gjrkt-MuiListSubheader-root"
          >
            <svg
              data-testid="logo-sidebar"
              height="20"
              version="1.1"
              viewBox="0 0 673 116"
              width="119"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                logo-hzn
              </title>
              <g
                fill="none"
                fill-rule="evenodd"
                id="logo-hzn"
                stroke="none"
                stroke-width="1"
              >
                <polyline
                  fill="#1A1919"
                  id="Fill-1"
                  points="204.818 17.049 226.366 16.956 226.74 101.699 205.192 101.794 205.047 69.228 181.197 69.331 181.342 101.898 159.671 101.993 159.298 17.25 180.968 17.153 181.106 48.509 204.955 48.405 204.818 17.049"
                />
                <path
                  d="M264.882,75.684 C269.255,79.983 274.714,82.118 281.251,82.0892915 C287.789,82.06 293.228,79.877 297.568,75.539 C301.907,71.206 304.062,65.689 304.030351,58.989 C304.002,52.29 301.799,46.791 297.421,42.489 C293.045,38.193 287.589,36.058 281.051,36.0867085 C274.513,36.115 269.072,38.299 264.735,42.634 C260.395,46.971 258.24,52.49 258.268711,59.189 C258.301,65.888 260.503,71.387 264.882,75.684 Z M312.826,90.364 C304.267,98.916 293.772,103.223 281.345,103.275551 C268.915,103.331 258.385,99.118 249.753,90.642 C241.119,82.163 236.777,71.713 236.720511,59.283 C236.668,46.855 240.919,36.363 249.476,27.812 C258.035,19.259 268.528,14.955 280.957,14.8995058 C293.384,14.847 303.917,19.057 312.549,27.534 C321.183,36.012 325.526,46.465 325.579527,58.893 C325.634,71.322 321.384,81.812 312.826,90.364 L312.826,90.364 Z"
                  fill="#1A1919"
                  id="Fill-2"
                />
                <path
                  d="M357.199,36.359 L357.281,55.365 L369.266,55.312 C371.928,55.381 374.107,54.524 375.793,52.74 C377.476,50.96 378.314,48.653 378.302127,45.829 C378.29,43.005 377.432,40.709 375.731,38.939 C374.03,37.173 371.846,36.294 369.184,36.3058788 L357.199,36.359 Z M379.998,101.024 L365.111,73.73 L357.363,73.766 L357.484,101.126 L335.813,101.22 L335.44,16.476 L369.337,16.325649 C377.969,16.29 385.225,19.023 391.1,24.527 C396.975,30.028 399.935,37.017 399.973,45.492 C399.994,50.576 398.72,55.242 396.158,59.493 C393.594,63.738 390.098,67.127 385.668,69.645 L403.243,100.924 L379.998,101.024 L379.998,101.024 Z"
                  fill="#1A1919"
                  id="Fill-3"
                />
                <polyline
                  fill="#1A1919"
                  id="Fill-4"
                  points="410.496 16.146 432.167 16.052 432.541 100.796 410.87 100.89 410.496 16.146"
                />
                <polyline
                  fill="#1A1919"
                  id="Fill-5"
                  points="471.172 79.682 502.043 79.547 502.137 100.49 443.192 100.748 443.117 84.041 472.813 36.819 443.515 36.947 443.423 16.001 500.8 15.751 500.874 32.453 471.172 79.682"
                />
                <path
                  d="M533.149,74.505 C537.523,78.805 542.981,80.939 549.518,80.9132347 C556.056,80.884 561.494,78.701 565.835,74.363 C570.175,70.028 572.329,64.51 572.29833,57.811 C572.269,51.111 570.067,45.612 565.688,41.313 C561.312,37.017 555.856,34.881 549.319,34.9106884 C542.78,34.937 537.339,37.12 533.002,41.456 C528.661,45.793 526.507,51.312 526.535711,58.01 C526.568,64.71 528.77,70.209 533.149,74.505 Z M581.093,89.186 C572.533,97.74 562.039,102.044 549.612,102.100476 C537.183,102.152 526.652,97.94 518.02,89.463 C509.386,80.987 505.043,70.534 504.987511,58.105 C504.935,45.677 509.186,35.188 517.743,26.633 C526.303,18.081 536.795,13.777 549.224,13.7234683 C561.652,13.669 572.185,17.881 580.817,26.356 C589.45,34.834 593.793,45.29 593.846546,57.716 C593.902,70.147 589.651,80.634 581.093,89.186 L581.093,89.186 Z"
                  fill="#1A1919"
                  id="Fill-6"
                />
                <polyline
                  fill="#1A1919"
                  id="Fill-7"
                  points="650.435 15.092 671.983 14.999 672.356 99.743 655.649 99.815 625.561 57.334 625.749 99.947 604.08 100.042 603.706 15.297 620.531 15.225 650.622 57.707 650.435 15.092"
                />
                <polyline
                  fill="#D8232A"
                  id="Fill-8"
                  points="114.355 0.999 45.728 0.999 45.728 46.996 68.592 46.996 68.592 23.863 91.49 23.863 91.49 116 114.355 116 114.355 0.999"
                />
                <polyline
                  fill="#D8232A"
                  id="Fill-9"
                  points="0 116 68.628 116 68.628 70.004 45.763 70.004 45.763 93.136 22.865 93.136 22.865 0.999 0 0.999 0 116"
                />
              </g>
            </svg>
          </div>
          <a
            class="MuiListItemButton-root MuiListItemButton-gutters Mui-selected MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
            href="/"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
            >
              <span
                aria-hidden="true"
                class="material-icons notranslate MuiIcon-root MuiIcon-fontSizeSmall css-1yybaem-MuiIcon-root"
                data-testid="icon-dashboard"
              >
                dashboard
              </span>
            </div>
            <div
              class="MuiListItemText-root css-tlelie-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
              >
                Dashboard
              </span>
            </div>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </a>
          <div
            class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
            >
              <span
                aria-hidden="true"
                class="material-icons notranslate MuiIcon-root MuiIcon-fontSizeSmall css-1yybaem-MuiIcon-root"
                data-testid="icon-people"
              >
                people
              </span>
            </div>
            <div
              class="MuiListItemText-root css-tlelie-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
              >
                Users
              </span>
            </div>
            <span
              aria-hidden="true"
              class="material-icons notranslate MuiIcon-root MuiIcon-fontSizeSmall css-1yybaem-MuiIcon-root"
              data-testid="icon-menu-1"
            >
              keyboard_arrow_down
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </div>
          <div
            class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
              >
                <div
                  class="MuiList-root css-1mk9mw3-MuiList-root"
                >
                  <a
                    class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
                    href="/users/list"
                    tabindex="0"
                  >
                    <div
                      class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                      >
                        List
                      </span>
                    </div>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </a>
                  <a
                    class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
                    href="/users/create"
                    tabindex="0"
                  >
                    <div
                      class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                      >
                        Create
                      </span>
                    </div>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </a>
                </div>
              </div>
            </div>
          </div>
        </nav>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        />
        <ul
          class="MuiList-root MuiList-padding css-n8txy3-MuiList-root"
        >
          <li
            class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
          >
            <a
              class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
              href="/settings"
              tabindex="0"
            >
              <div
                class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
              >
                <span
                  aria-hidden="true"
                  class="material-icons notranslate MuiIcon-root MuiIcon-fontSizeSmall css-1yybaem-MuiIcon-root"
                  data-testid="icon-settings"
                >
                  settings
                </span>
              </div>
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                >
                  Settings
                </span>
              </div>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
          </li>
        </ul>
        <div
          class="MuiBox-root css-19kzrtu"
        >
          <button
            class="MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-fullWidth MuiButtonBase-root  css-15t7wz5-MuiButtonBase-root-MuiButton-root"
            data-testid="button-logout"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeSmall css-y6rp3m-MuiButton-startIcon"
            >
              <span
                aria-hidden="true"
                class="material-icons-outlined notranslate MuiIcon-root MuiIcon-fontSizeMedium css-kp9ftd-MuiIcon-root"
                data-testid="icon-sidebar-logout"
              >
                logout
              </span>
            </span>
            Logout
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <main
      class="MuiBox-root css-m4zwgh"
      data-testid="layout-default-test"
    >
      Children component
    </main>
  </div>
</div>
`;
